code/modules/admin/verbs/randomverbs.dm

```diff
@@ -1190,3 +1190,57 @@ Traitors and the like can also be revived with the previous role mostly intact.
 					if(!source)
 						return
 			REMOVE_TRAIT(D, chosen_trait, source)
+
+/client/proc/create_crate(object as text)
+	set name = "Create Crate"
+	set desc = "Spawn a crate from a supplypack datum. Append a period to the text in order to exclude subtypes of paths matching the input."
+	set category = "Event"
+
+	if(!check_rights(R_SPAWN))
+		return
+
+	var/list/types = list()
+	for(var/typepath in subtypesof(/datum/supply_packs))
+		var/datum/supply_packs/P = typepath
+		if(initial(P.name) == "HEADER")
+			continue // To filter out group headers
+		types += P
+
+	var/list/matches = list()
+
+	var/include_subtypes = TRUE
+	if(copytext(object, -1) == ".")
+		include_subtypes = FALSE
+		object = copytext(object, 1, -1)
+
+	if(include_subtypes)
+		for(var/path in types)
+			if(findtext("[path]", object))
+				matches += path
+	else
+		var/needle_length = length(object)
+		for(var/path in types)
+			if(copytext("[path]", -needle_length) == object)
+				matches += path
+
+	if(!length(matches))
+		return
+
+	var/chosen
+	if(length(matches) == 1)
+		chosen = matches[1]
+	else
+		chosen = input("Select a supply crate type", "Create Crate", matches[1]) as null|anything in matches
```

Comment:
```suggestion
	var/chosen = input("Select a supply crate type", "Create Crate", matches[1]) as null|anything in matches
```
Just for the sake of admins having much more control over the tool (and not being as prone to mistakes), its best that they can A) see that they have the right crate before confirming and B) always have the opportunity to cancel if they didn't get the right type they were intending for